### PR TITLE
Feature-10: Fixed issues in Pull Request

### PR DESCRIPTION
**Pull request by kudokuna**
I've done next activities:
1) Refactored packages structure according to new order
2) Completed class work for lesson2
3) Completed hometask1
4) Completed hometask2
5) Fixed issues in Pull Request

### CODE REVIEW

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.Scanner;
+
+public class Main {
+
+    public static void main(String[] args) {
+        rectangleCalculation();
+    }
+
+    public static void rectangleCalculation()
+    {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java]: Better to start class, constructors and methods body on the same line as the name.

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        double side1;
+        double side2;
+        Scanner sc = new Scanner(System.in);
+
+        while (true)
+        {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java]: At the same line.

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            System.out.println("Hi, type first value");
+            if (sc.hasNextDouble())
+            {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java]: At the same line.

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                side1 = sc.nextDouble();
+                System.out.println("Hi, type second value");
+                if (sc.hasNextDouble())
+                {
+                    side2 = sc.nextDouble();
+                    break;
+                }
+            }
+            System.out.println("Wrong input");
+
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java:19]: CTRL + ALT + L

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Rectangle.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    public Rectangle (double side1, double side2)
+    {
+        this.side1 = side1;
+        this.side2 = side2;
+    }
+
+    double areaCalculator (double side1, double side2)
+    {
+        double area = side1 * side2;
+        return  area;
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Rectangle.java]: The variable "area" is not needed, just return the result of the operation(return  side1 * side2).

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Rectangle.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    double perimeterCalculator (double side1, double side2)
+    {
+        double perimeter = (side1 + side2) * 2;
+        return perimeter;
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Rectangle.java]: Return the result of the operation.

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Book.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package com.java_essential.kopaiev.essential.homework.lesson1.task3;
+
+import java.awt.desktop.SystemSleepEvent;
+
+public class Book {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Book.java:7]: The Book class must have placeholders for author, title, and content.
+
It should have a "show" method that prints the author, title, and content to the console.

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task4/Computer.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package com.java_essential.kopaiev.essential.homework.lesson1.task4;
+
+public class Computer {
+    public static void main(String[] args) {
+
+        Object[] computers = new Object[5];
+
+        for (int i = 0; i < computers.length; i++) {
+            computers[i] = new Computer();
+        }
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task4/Computer.java:22]: Create five instance of Computer by HANDS, than, with loop, add them to the list)))

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson2/task2/Car.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package com.java_essential.kopaiev.essential.homework.lesson2.task2;
+
+public class Car {
+    private int year;
+    private String color;
+
+    public Car()
+    {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson2/task2/Car.java]: Please, don't do that)

public Car() {}

**Review by ochedenis — CHANGES_REQUESTED** [review]

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java (hunk 1 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java]: fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java (hunk 2 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java]: fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java (hunk 3 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java]: fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java (hunk 4 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Main.java:19]: fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Rectangle.java (hunk 5 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Rectangle.java]: fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Rectangle.java (hunk 6 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task2/Rectangle.java]: fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Book.java (hunk 7 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Book.java:7]: Change logic for Book class. Please check

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task4/Computer.java (hunk 8 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task4/Computer.java:22]: Probably fixed. Changed according to approved Pull Request

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson2/task2/Car.java (hunk 9 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson2/task2/Car.java]: fixed

**Review by yuriy-tsarenko — APPROVED** [review]

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Book.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+    public void setContent(Content content) {
+        this.content = content;
+    }
+
+    Title title = new Title();
+    Content content = new Content();
+
+    public Book(Author author, Title title, Content content) {
+
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Book.java:18]: Why this constructor empty?
All logic at "fillData" method you can use thru this constructor.

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Content.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    public String getContentName() {
+        return contentName;
+    }
+
+    public void setContentName(String contentName) {
+        this.contentName = contentName;
+    }
+
+    public void show()
+    {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Content.java]: At same line as name.

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Title.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    public String getTitleName() {
+        return titleName;
+    }
+
+    public void setTitleName(String titleName) {
+        this.titleName = titleName;
+    }
+
+    public void show()
+    {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Title.java]: At same line as name.

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Author.java [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+    public String getAuthorName() {
+        return authorName;
+    }
+
+    public void setAuthorName(String authorName) {
+        this.authorName = authorName;
+    }
+
+    public void show()
+    {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Author.java]: At same line as name.

**Review by ochedenis — CHANGES_REQUESTED** [review]

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Content.java (hunk 11 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Content.java]: Fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Title.java (hunk 12 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Title.java]: Fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Author.java (hunk 13 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Author.java]: Fixed

File: src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Book.java (hunk 10 above)
Comment (kudokuna) [reply src/main/java/com/java_essential/kopaiev/essential/homework/lesson1/task3/Book.java:18]: Fixed. Removed fillData method.

**Review by yuriy-tsarenko — COMMENTED** [review]
image

add gitignore file to project and exclude folder "target" from git index, file format like ".class" is a bad practice

**Review by yuriy-tsarenko — COMMENTED** [review]
exclude folder "target" from index
use the next git commands:
$ git reset head
$ git add .

**Review by yuriy-tsarenko — COMMENTED** [review]
and commit changes

**Pull request merged**